It looks like our linter is giving errors about the standard library:

```
01:47:56 ../../../../../usr/local/go/src/os/user/lookup.go:38:9:warning: undeclared name: listGroups (unconvert)
01:47:56 ../../../../../usr/local/go/src/os/user/lookup.go:9:9:warning: undeclared name: current (unconvert)
01:47:56 ../../../../../usr/local/go/src/os/user/lookup.go:15:9:warning: undeclared name: lookupUser (unconvert)
01:47:56 ../../../../../usr/local/go/src/os/user/lookup.go:21:9:warning: undeclared name: lookupUserId (unconvert)
01:47:56 ../../../../../usr/local/go/src/os/user/lookup.go:27:9:warning: undeclared name: lookupGroup (unconvert)
01:47:56 ../../../../../usr/local/go/src/os/user/lookup.go:33:9:warning: undeclared name: lookupGroupId (unconvert)
```

--- a/daemon/graphdriver/devmapper/deviceset.go
+++ b/daemon/graphdriver/devmapper/deviceset.go
@@ -2416,6 +2416,18 @@ func (devices *DeviceSet) UnmountDevice(hash, mountPath string) error {
 	}
 	logrus.Debug("devmapper: Unmount done")
 
+	// Remove the mountpoint here. Removing the mountpoint (in newer kernels)
+	// will cause all other instances of this mount in other mount namespaces
+	// to be killed (this is an anti-DoS measure that is necessary for things
+	// like devicemapper). This is necessary to avoid cases where a libdm mount
+	// that is present in another namespace will cause subsequent RemoveDevice
+	// operations to fail. We ignore any errors here because this may fail on
+	// older kernels which don't have
+	// torvalds/linux@8ed936b5671bfb33d89bc60bdcc7cf0470ba52fe applied.
+	if err := os.Remove(mountPath); err != nil {
+		logrus.Debugf("devmapper: error doing a remove on unmounted device %s: %v", mountPath, err)
+	}
+
 	return devices.deactivateDevice(info)
 }
 

--- a/daemon/graphdriver/devmapper/driver.go
+++ b/daemon/graphdriver/devmapper/driver.go
@@ -228,10 +228,12 @@ func (d *Driver) Put(id string) error {
 	if count := d.ctr.Decrement(mp); count > 0 {
 		return nil
 	}
+
 	err := d.DeviceSet.UnmountDevice(id, mp)
 	if err != nil {
-		logrus.Errorf("devmapper: Error unmounting device %s: %s", id, err)
+		logrus.Errorf("devmapper: Error unmounting device %s: %v", id, err)
 	}
+
 	return err
 }
 

